gallery_dl/extractor/hatenablog.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://hatenablog.com"""
+
+import re
+from .common import Extractor, Message
+from .. import text
+
+
+BASE_PATTERN = (
+    r"(?:hatenablog:https?://([^/]+)|(?:https?://)?"
```

Comment:
```suggestion
    r"(?:hatenablog:https?://([^/?#]+)|(?:https?://)?"
```

This would otherwise match weird URLs like
```
hatenablog:https://example.org?key=value#extra/...
```